src/lines-yardstick.coffee

```diff
@@ -0,0 +1,164 @@
+LineHtmlBuilder = require './line-html-builder'
+AcceptFilter = {acceptNode: -> NodeFilter.FILTER_ACCEPT}
+rangeForMeasurement = document.createRange()
+{Emitter} = require 'event-kit'
+{last} = require 'underscore-plus'
+
+module.exports =
+class LinesYardstick
+  constructor: (@editor, @presenter, hostElement, @syntaxStyleElement) ->
+    @initialized = false
+    @emitter = new Emitter
+    @linesBuilder = new LineHtmlBuilder(true)
+    @stylesNode = document.createElement("style")
+    @iframe = document.createElement("iframe")
+    @iframe.style.display = "none"
+    @iframe.onload = @setupIframe
+    @lineNodesByScreenRow = {}
+    @lineNodesByContextIndex = {}
+    @screenRowsByLineId = {}
+
+    hostElement.appendChild(@iframe)
+
+  setFont: (fontFamily, fontSize) ->
+    @stylesNode.innerHTML = "body { margin: 0; padding: 0; font-size: #{fontSize}; font-family: #{fontFamily}; white-space: pre; }"
+
+  onDidInitialize: (callback) ->
+    @emitter.on "did-initialize", callback
+
+  canMeasure: ->
+    @initialized
+
+  ensureInitialized: ->
+    return if @canMeasure()
+
+    throw new Error("This instance of LinesYardstick hasn't been initialized!")
+
+  createMeasurementContext: ->
+    context = document.createElement("div")
+    context.style.overflow = "hidden"
+    context.style.width = "600px"
+    context.style.height = "600px"
+    context.style.display = "block"
+    context
+
+  setupIframe: =>
+    @initialized = true
+    @domNode = @iframe.contentDocument.body
+    @headNode = @iframe.contentDocument.head
+    @headNode.appendChild(@stylesNode)
+    @headNode.appendChild(@syntaxStyleElement)
```

Comment:
Here I am adding the whole `<style>` element, but with a little effort we may generate a minimal CSS document that contains only those selectors with font informations.

Similarly, we can reduce scope HTML complexity by dropping those scopes that don’t have a matching CSS selector in the previously generated CSS.